packages/peregrine/lib/store/actions/checkout/asyncActions.js

```diff
@@ -113,35 +95,41 @@ export const getShippingMethods = () => {
 
 export const submitPaymentMethodAndBillingAddress = payload =>
     async function thunk(dispatch, getState) {
-        submitBillingAddress(payload.formValues.billingAddress)(
-            dispatch,
-            getState
-        );
-        submitPaymentMethod(payload.formValues.paymentMethod)(
-            dispatch,
-            getState
-        );
+        const { countries, formValues } = payload;
+        const { billingAddress, paymentMethod } = formValues;
+
+        submitBillingAddress({
+            billingAddress,
+            countries
+        })(dispatch, getState);
```

Comment:
This syntax looks odd. This is an async action. We should probably have this:

```js
await Promise.all([
  dispatch(submitBillingAddress({ billingAddress, countries })),
  dispatch(submitPaymentMethod(paymentMethod));
]);
```

Reply:
💪 .

Omitted the `Promise.all` because the previous functionality wasn't waiting.

https://github.com/magento/pwa-studio/pull/1993/commits/5771d0717f5ed71387c30784260699dd86a3de2a